src/cpp/session/modules/SessionCodeSearch.cpp

```diff
@@ -110,12 +111,32 @@ bool isWithinIgnoredDirectory(const FilePath& filePath)
       // websites
       if (parentName == websiteDir)
          return true;
+
+      // quarto projects
+      if (parentName == quartoProjectOutputDir)
+         return true;
       
       // packrat
       if (parentName == "packrat" &&
          parentPath.completeChildPath("packrat.lock").exists())
          return true;
       
+      // packrat
+      if (parentName == "packrat" &&
+         parentPath.completeChildPath("packrat.lock").exists())
+         return true;
+
+      // renv
+      if (parentName == "renv" &&
+         parentPath.completeChildPath("settings.dcf").exists())
```

Comment:
It's possible for a "bare" `renv` environment to not have any project settings defined, so I think this is a better check:

```suggestion
         parentPath.completeChildPath("activate.R").exists())
```